### PR TITLE
Refactor course page

### PR DESCRIPTION
**Pull request by c-neli-r**
# Description of changes
- Refactor course page: different tabs have now their own url e.g /forum, /leaderboard, .... --> this allows navigating in different tabs from different places of the application for example we can now navigate from the profile to forum thread or from the thread to the related media
- Other changes:
- Answer to to threads
- Mark other course users with @
- Delete threads
- Select and unselect best answer
- Refactor resizable forum and media
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [ ] My code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/MEITREX/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: components/forum/richTextEditor/MentionSuggestion.tsx
Comment (ZiyaSelim) [inline components/forum/richTextEditor/MentionSuggestion.tsx:1]: Sollte man sich selber metionen können? Weil in nem Kurs/Thread nur mit mir selber gab es keine users in der Mentionlist als ichs probiert habe
grafik

File: components/forum/shared/ForumActivity.tsx [hunk 1]
```diff
@@ -28,11 +32,28 @@ export default function ForumActivity({
...
+       3. Profile page other user
+  */
+  const navigate = () => {
+    if (pathname.includes("/courses")) {
+      router.push(`${pathname}/forum/${thread.id}`);
+    } else if (pathname.includes("/profile")) {
+      router.push(`/courses/${courseId}/forum/${thread.id}`);
+    }
+  };
+
```
Comment (ZiyaSelim) [inline components/forum/shared/ForumActivity.tsx:53]: Wo wäre hier der path für den dritten bpunkt? Also das profil eines anderen Nutzers

**Review by Alexander848 — APPROVED** [review]
